product_last_purchase_date/tests/test_product_last_purchase_date.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+from datetime import timedelta
+
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestProductLastPurchaseDate(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestProductLastPurchaseDate, cls).setUpClass()
+        cls.product_template = cls.env["product.template"].create(
+            {"name": "Test Product", "type": "product"}
+        )
+        cls.product = cls.product_template.product_variant_ids
+        cls.location = cls.env.ref("stock.stock_location_suppliers")
+        cls.location_dest = cls.env.ref("stock.stock_location_stock")
+
+    def create_incoming_receipt(self):
+        picking = self.env["stock.picking"].create(
+            {
+                "location_id": self.location.id,
+                "location_dest_id": self.location_dest.id,
+                "picking_type_id": self.env.ref("stock.picking_type_in").id,
+                "move_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "Test Move",
+                            "location_id": self.location.id,
+                            "location_dest_id": self.location_dest.id,
+                            "product_id": self.product.id,
+                            "product_uom": self.product.uom_id.id,
+                            "product_uom_qty": 1,
+                        },
+                    )
+                ],
+            }
+        )
+        picking.action_confirm()
+        picking.action_assign()
+        picking.button_validate()
```

Comment:
Is this step necessary?

Reply:
Yes. It is because of this domain https://github.com/qrtl/axls-custom/blob/60cf21b44348191492282bbef209b63d1df0ac7e/product_last_purchase_date/models/product_product.py#L45.